### PR TITLE
Add Hadoop LZOP

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

**Comment by matt-martin** [comment]
Thanks for putting this together @dain.  Is there anything we can do to help make sure this makes its way into Presto?

**Comment by dain** [comment]
I'm waiting on a review from @martint

**Comment by Acehaidrey** [comment]
Hey all,
Wondering what the status of this patch is. Ran into similar issues myself, and want to know if this will be available in the next presto release.

**Comment by kmarutya** [comment]
Waiting on resolution for lzo support as well. Is there any timeline for this?
Thanks in advance.

File: src/main/java/io/airlift/compress/lzo/HadoopLzopInputStream.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+            }
+            readInput(compressed, 0, compressedLength);
+            int actualUncompressedLength = decompressor.decompress(compressed, 0, compressedLength, output, outputOffset, outputLength);
+            if (actualUncompressedLength != uncompressedLength) {
+                throw new IOException("Decompressor did not decompress the entire block");
+            }
+        }
+    }
+
+    private void readInput(byte[] buffer, int offset, int length)
```
Comment (martint) [inline src/main/java/io/airlift/compress/lzo/HadoopLzopInputStream.java]: Maybe call it readFully

**Comment by matt-martin** [comment]
Thank you for merging this in @dain!  Is there any chance we can release a new version of aircompressor and bump the dependency version in presto to close out https://github.com/prestodb/presto/issues/7348?

**Comment by dain** [comment]
We are working on a release now.  This should be able to go into the next release.

**Comment by matt-martin** [comment]
Awesome. Thanks!

**Pull request merged**